### PR TITLE
Added support Sec-CH-UA-Form-Factors header for ClientHints

### PR DESCRIPTION
**Pull request by sanchezzzhak**
Specs: https://wicg.github.io/ua-client-hints/#sec-ch-ua-form-factors
form factor support for chrome 124+  https://chromestatus.com/feature/5162545698045952

### CODE REVIEW

**Comment by sanchezzzhak** [comment]
maybe we should add examples to the documentation for client-hints?

For example, through JS, we can get them like this:

```js
function getHints() {
    return new Promise((resolve, reject) => {
      if (!navigator.userAgentData) {
        return resolve({})
      }
      let requestHints = [
        'brands',
        'mobile',
        'platform',
        'platformVersion',
        'architecture',
        'bitness',
        'wow64',
        'model',
        'formFactors',
        'uaFullVersion',
        'fullVersionList'];

      navigator.userAgentData.getHighEntropyValues(requestHints).then((result) => {
        resolve(JSON.parse(JSON.stringify(result)))
      }).catch(() => {
        resolve({});
      });
    })
  }

```

```js
getHints().then(data =>console.log(data)) 
```

**Comment by sgiehl** [comment]
@sanchezzzhak Based on the definition formFactors is a `sf-list`. So it can contain multiple comma separated values. In theory something like `"Tablet","EInk"` would be a valid value. So we would actually need to handle that as array and not as string. Or did I misinterpret the definition 🤔

**Comment by liviuconcioiu** [comment]
For reference, `Android` tablets are reporting as `Desktop`, but detection works.

tablets

**Comment by sanchezzzhak** [comment]
@sgiehl yes there should be an array.

---

I think the form factor based on the user agent string, since there is no prefix (Table|Mobile|VR|TV|XR) is set by default to Desktop

File: Tests/ClientHintsTest.php [hunk 1]
```diff
@@ -47,6 +47,7 @@ public function testHeadersHttp(): void
             'HTTP_SEC_CH_UA_PLATFORM'          => 'Ubuntu',
             'HTTP_SEC_CH_UA_PLATFORM_VERSION'  => '3.7',
             'HTTP_SEC_CH_UA_FULL_VERSION'      => '98.0.14335.105',
+            'HTTP_SEC-CH-UA-FORM-FACTORS'      => '"Desktop"',
```
Comment (liviuconcioiu) [inline Tests/ClientHintsTest.php:50]:
```suggestion
            'HTTP_SEC_CH_UA_FORM_FACTORS'      => '"Desktop"',
```

**Review by sgiehl — APPROVED** [review]
Looks good to me now.

**Pull request merged**